### DISCUSSION TITLE
Target-typed interfaces in anonymous types

### DISCUSSION THREAD

**Discussion by RenderMichael** [dotnet/csharplang#7893] [Language Ideas] [answered]
Anonymous types add a lot of convenience when you just want an ad-hoc type quickly.

Allowing anonymous types to meet interface criteria based on the target of the type, sounds like a nice convenience.

``` csharp
M(new { Context = "" }); // this could be made work

void M(I i) { }

public interface I
{
    string Context { get; }
}
```

**Comment by TahirAhmadov** [comment, marked as answer]
https://github.com/dotnet/csharplang/issues/4301
https://github.com/dotnet/csharplang/discussions/7719
https://github.com/dotnet/csharplang/discussions/7726

**Comment by DavidArno** [comment]
This already works, so doesn't require a language change, and does exactly what you want in fewer lines of code:
```cs
M(new(Context = "" )); 

void M(R r) { }

public record R(string Context);
```

**Reply by colejohnson66** [reply]
That's a record, not an interface

**Reply by RenderMichael** [reply]
This resembles my minimal repro but it can’t do what this feature could.

**Comment by DavidArno** [comment]
> This resembles my minimal repro but it can’t do what this feature could.

I'll take your word for it 😃
